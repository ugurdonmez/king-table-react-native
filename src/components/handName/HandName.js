import React from 'react';
import PropTypes from 'prop-types';
import { Text, StyleSheet } from 'react-native'

const HandName = (props) => {
    return (
        <Text style={styles.container}>
            {props.name}
        </Text>
    );
}

HandName.propTypes = {
    name: PropTypes.string,
}

export default HandName;

var styles = StyleSheet.create({
    container: {
        textAlign: 'center',
        borderRadius: 4,
        borderWidth: 0.5,
        borderColor: '#d6d7da',
        width: '20%',
    },
})
