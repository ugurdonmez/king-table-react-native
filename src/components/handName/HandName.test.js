import React from 'react';
import 'react-native';
import HandName from './HandName';
import renderer from 'react-test-renderer'

describe('<HandName />', () => {
    it('renders tree without crashing', () => {
        const tree = renderer.create(
            <HandName
                name="El Almaz"
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })

    it('text should be equal to prop name', () => {
        const wrapper = renderer.create(
            <HandName 
                name="El Almaz"
            />
        ).toJSON();

        expect(wrapper.children[0]).toBe('El Almaz');
    })
})