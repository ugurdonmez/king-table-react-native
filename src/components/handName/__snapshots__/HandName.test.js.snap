// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HandName /> match snapshot 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "borderColor": "#d6d7da",
      "borderRadius": 4,
      "borderWidth": 0.5,
      "textAlign": "center",
      "width": "20%",
    }
  }
>
  El Almaz
</Text>
`;

exports[`<HandName /> renders tree without crashing 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "borderColor": "#d6d7da",
      "borderRadius": 4,
      "borderWidth": 0.5,
      "textAlign": "center",
      "width": "20%",
    }
  }
>
  El Almaz
</Text>
`;
